activiti-core/activiti-spring-process-extensions/src/test/java/org/activiti/spring/process/model/ExtensionTest.java

```diff
@@ -34,33 +35,52 @@
     @BeforeEach
     public void setUp() {
         initMocks(this);
-        given(processVariablesMapping.getInputs()).willReturn(emptyMap());
-        given(processVariablesMapping.getOutputs()).willReturn(emptyMap());
     }
 
     @Test
-    public void hasEmptyInputsMappingShouldReturnTrueWhenInputsMapIsEmpty() {
+    public void should_bothHasMappingTypeInputsAndOutputsReturnTrue_when_thereIsMappingTypeMAP_ALL() {
+        given(processVariablesMapping.getMappingType()).willReturn(MappingType.MAP_ALL);
+
         Extension extension = new Extension();
+        HashMap<String, ProcessVariablesMapping> mapping = new HashMap<>();
+        mapping.put("elementId", processVariablesMapping);
+        extension.setMappings(mapping);
+
+        assertThat(extension.hasMappingTypeInputs("elementId")).isTrue();
+        assertThat(extension.hasMappingTypeOutputs("elementId")).isTrue();
+
+    }
 
+    @Test
+    public void should_onlyHasMappingTypeInputsReturnTrue_when_thereIsMappingTypeMAP_ALL_INPUTS() {
+        given(processVariablesMapping.getMappingType()).willReturn(MappingType.MAP_ALL_INPUTS);
+
+        Extension extension = new Extension();
         HashMap<String, ProcessVariablesMapping> mapping = new HashMap<>();
         mapping.put("elementId", processVariablesMapping);
         extension.setMappings(mapping);
 
-        assertThat(extension.hasEmptyInputsMapping("elementId")).isTrue();
+        assertThat(extension.hasMappingTypeInputs("elementId")).isTrue();
+        assertThat(extension.hasMappingTypeOutputs("elementId")).isFalse();
+
     }
 
     @Test
-    public void hasEmptyOutputsMappingShouldReturnTrueWhenOutputsMapIsEmpty() {
+    public void should_onlyHasMappingTypeOutputsReturnTrue_when_thereIsMappingTypeMAP_ALL_OUTPUTS() {
+        given(processVariablesMapping.getMappingType()).willReturn(MappingType.MAP_ALL_OUTPUTS);
+
         Extension extension = new Extension();
         HashMap<String, ProcessVariablesMapping> mapping = new HashMap<>();
         mapping.put("elementId", processVariablesMapping);
         extension.setMappings(mapping);
 
-        assertThat(extension.hasEmptyOutputsMapping("elementId")).isTrue();
+        assertThat(extension.hasMappingTypeInputs("elementId")).isFalse();
+        assertThat(extension.hasMappingTypeOutputs("elementId")).isTrue();
+
     }
 
     @Test
-    public void hasMappingShouldReturnTrueWhenThereIsMapping() {
+    public void should_hasMappingReturnTrue_when_thereIsMapping() {
```

Comment:
Personally, I prefer `hasMapping_should_returnTrue_when_thereIsMapping`: it's more fluent and closer to the speaking language.

Reply:
There is a naming convention for tests that we all agreed to, hence this naming. If you don't like that, let's change that, otherwise we should refrain from individual variants as we end up with methods in 100s of style

Reply:
Ok. It makes sense. Let's review the naming convention for tests to clarify some specific points.